code/datums/emotes.dm

```diff
@@ -143,6 +151,56 @@
 	user.emotes_used[src] = world.time
 	return TRUE
 
+/**
+ * Displays the emote_overlay (if the emote has one) on the user and removes it after the duration specified by the emote_overlay.
+ * Creates an instance of the emote_overlay and adds it to the emote_overlay_instances with the user as key.
+ *
+ * * Arguments:
+ * * user - Person that is trying to send the emote.
+ */
+/datum/emote/proc/display_overlay(mob/user)
+	if(!emote_overlay)
+		return
+	if(ishuman(user)) // Give them a visual effect if they're human
+		var/mob/living/carbon/human/human_user = user
+		var/obj/item/bodypart/attached_bodypart
+		var/datum/bodypart_overlay/simple/emote/overlay
+		if(user in emote_overlay_instances) // If we already have a visual for this user:
+			overlay = emote_overlay_instances[user] // Reference the existing one so we can still delete it later
+			attached_bodypart = human_user.get_bodypart(overlay.attached_body_zone)
+			if(!attached_bodypart)
+				return
+		else // Otherwise we make a new one and apply it to the user
+			overlay = new emote_overlay()
+			emote_overlay_instances[user] = overlay
```

Comment:
Not sure if they're singletons exactly but there's a global list with one instance of each emote. Using weakrefs worked though, and now I'm worried because I thought that it'd break with how I did it. It makes a new weakref to the user each time, but somehow weakrefs to the same user are recognized as the same key for emote_overlay_instances anyways